app/controllers/my_modules_controller.rb

```diff
@@ -450,6 +467,13 @@ def unassign_repository_records
       end
 
       @my_module.repository_rows.destroy(records & @my_module.repository_rows)
+
+      if downstream
```

Comment:
Same as with assign: this code should be rewritten because right now, if there is a row that is not assigned to the current task, it will also get skipped from downstream tasks.

By the way, the `&` operator in `records & my_module.repository_rows` simply performs the intersection of both arrays.